Tests/CommonTests/AssemblyResolverTests.cs

```diff
@@ -20,145 +20,244 @@ public class AssemblyResolverTests
-        }
+                // Assert
+                Assert.IsNotNull(simpleObject);
+                Assert.AreEqual<string>("SimpleProgram", simpleObject.GetType().ToString());
+                AssertResolverCaller(resolver);
 
-        private static Assembly CompileSimpleAssembly(string outputFilePath, ILogger logger)
-        {
-            return CompileAssembly(@"public class SimpleProgram {
-              public static void Main(string[] args) {
-                System.Console.WriteLine(""Hello World"");
-              }
-            }", outputFilePath, logger);
+            }
         }
 
-        #region tests
-
         /// <summary>
-        /// Tests that method for creating file names from assembly names is correct.
+        /// Tests the case where assembly resolution should fail correctly.
         /// </summary>
         [TestMethod]
-        public void TestAssemblyNameFileNameAssociation()
+        public void AssemblyResolver_NonExistentAssembly_ResolutionFails()
         {
             // Arrange
             TestLogger logger = new TestLogger();
-            Assembly assembly = typeof(AssemblyResolver).Assembly;
-            string assemblyName = assembly.FullName;
-            string actualFileName = Path.GetFileName(assembly.Location);
+            string testFolder = TestUtils.CreateTestDirectory(this.TestContext);
 
             // Act
-            string testFileName = AssemblyResolver.CreateFileNameFromAssemblyName(assemblyName);
+            using (AssemblyResolver resolver = new AssemblyResolver(logger, testFolder))
+            {
+                AssertAssemblyLoadFails("nonexistent library");
 
-            // Assert
-            Assert.AreEqual<string>(actualFileName, testFileName);
+                // Assert
+                AssertResolverCaller(resolver);
+            }
         }
 
         /// <summary>
-        /// Tests the loading of an assembly with a single type and no dependencies. This should succeed even without AssemblyResolver.
+        /// Tests the case where assembly resolution should succeed.
         /// </summary>
         [TestMethod]
-        public void TestNoImpactOnDefaultResolution()
+        public void AssemblyResolver_ResolutionByFullAssemblyName_Succeeds()
         {
             // Arrange
-            TestLogger logger = new TestLogger();
             string testFolder = TestUtils.CreateTestDirectory(this.TestContext);
-            string simpleAssemblyPath = Path.Combine(testFolder, "SimpleAssembly.dll");
-            Assembly simpleAssembly = CompileSimpleAssembly(simpleAssemblyPath, logger);
+            Assembly testAssembly = CompileSimpleAssembly("SimpleAssemblyByFullName.dll", testFolder, new TestLogger());
 
-            object simpleObject = null;
+            // Act
+            Assembly resolvedAssembly = AssertAssemblyLoadSucceedsOnlyWithResolver("SimpleAssemblyByFullName, Version = 0.0.0.0, Culture = neutral, PublicKeyToken = null", testFolder);
```

Comment:
Test loading using the assembly name format i.e. "name, version=xxx, culture=xxx..." etc